apps/stable_diffusion/src/utils/stencils/zoe/__init__.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import torch
+from pathlib import Path
+import requests
+
+
+from einops import rearrange
+from .models.zoedepth.zoedepth_v1 import ZoeDepth
+from .utils.config import get_config
+
+remote_model_path = (
+    "https://huggingface.co/lllyasviel/Annotators/resolve/main/ZoeD_M12_N.pt"
+)
+
+
+class ZoeDetector:
+    def __init__(self):
+        cwd = Path.cwd()
+        ckpt_path = Path(cwd, "stencil_annotator")
+        ckpt_path.mkdir(parents=True, exist_ok=True)
+        modelpath = ckpt_path / "ZoeD_M12_N.pt"
+
+        with requests.get(remote_model_path, stream=True) as r:
+            r.raise_for_status()
+            with open(modelpath, "wb") as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+
+        if not modelpath.is_file():
+            midas = torch.hub.load(
+                "gpetters94/MiDaS",
+                "DPT_BEiT_L_384",
+                pretrained=False,
+                force_reload=False,
+            )
+            zoe = torch.hub.load(
+                "AyaanShah2204/MiDaS",
```

Comment:
@gpetters94 why are we calling the MiDaS model twice? Did you mean `AyaanShah2204/ZoeDepth`?

Reply:
Also, I can modify `AyaanShah2204/ZoeDepth` to internally load `Ayaanshah2204/MiDaS` instead of the `intel-org` version. Then we don't need to worry about `torch.load` potentially caching conflicting models.

Reply:
> Also, I can modify `AyaanShah2204/ZoeDepth` to internally load `Ayaanshah2204/MiDaS` instead of the `intel-org` version. Then we don't need to worry about `torch.load` potentially caching conflicting models.

That'd be great, that was causing some issues. If you could also rebase your MiDaS version with mine we can switch yours back. It's the fix you were talking about in the MiDaS repo (the one Illyasviel did) that fixed the version issues with TOM SHARK.